typegate/tests/metagen/typegraphs/identities.py

```diff
@@ -0,0 +1,198 @@
+from typegraph import typegraph, Policy, t, Graph
+from typegraph.runtimes.python import PythonRuntime
+from typegraph.runtimes.wasm import WasmRuntime
+from typegraph.runtimes.deno import DenoRuntime
+
+
+@typegraph()
+def identities(g: Graph):
+    # TODO: injections
+    # TODO: materializers (HttpRuntime.rest)
+
+    primitives = t.struct(
+        {
+            "str": t.string(),
+            "enum": t.enum(["wan", "tew", "tree"]),
+            "uuid": t.uuid(as_id=True),
+            "email": t.email(),
+            "ean": t.ean(),
+            "json": t.json(),
+            "uri": t.uri(),
+            "date": t.date(),
+            "datetime": t.datetime(),
+            "int": t.integer(),
+            "float": t.float(),
+            "boolean": t.boolean(),
+            # TODO: file upload support for MDK?
+            # "file": t.file(),
+        }
+    ).rename("primitives")
+    composites = t.struct(
+        {
+            "opt": t.optional(t.string()),
+            "either": t.either(
+                [
+                    primitives,
+                    t.struct({"branch2": t.string()}).rename("branch2"),
+                ]
+            ),
+            "union": t.union(
+                [
+                    t.list(t.enum(["grey", "beige"])).rename("branch4"),
+                    t.integer(),
+                    t.string(),
+                    t.email().rename("branch4again"),
+                ]
+            ),
+            "list": t.list(t.string()),
+        }
+    ).rename("composites")
+    cycles1 = t.struct(
```

Comment:
These are the cyclical types that break the typegate.

Reply:
Does it break at validation or.. ?

Reply:
It 500s the typegate.  Nested union/either should have been flattened.